libr/core/cmd_anal.c

```diff
@@ -976,21 +976,38 @@ static void var_help(RCore *core, char ch) {
 	}
 }
 
-static void var_accesses_list(RAnalFunction *fcn, RAnalVar *var, int access_type, const char *name) {
+static void var_accesses_list(RAnalFunction *fcn, RAnalVar *var, PJ *pj, int access_type, const char *name) {
 	RAnalVarAccess *acc;
 	bool first = true;
-	r_cons_printf ("%10s", name);
+	if (!pj) {
+		r_cons_printf ("%10s", name);
+	}
 	r_vector_foreach (&var->accesses, acc) {
 		if (!(acc->type & access_type)) {
 			continue;
 		}
-		r_cons_printf ("%s0x%"PFMT64x, first? "  ": ",", (ut64)((st64)fcn->addr + acc->offset));
+		ut64 addr = fcn->addr + acc->offset;
+		if (pj && first) {
```

Comment:
why does it create a new json object on the first element and then never closes it? i would have expected that all variable accesses were in a array.

Reply:
@ret2libc I don't know why I did that, fixed it 4e1e11a24b5f320e49aa350d3254ec3bf199fefe